bluesky/run_engine.py

```diff
@@ -298,13 +298,25 @@ def __init__(self, md=None, *, loop=None, preprocessors=None,
         self._th = _ensure_event_loop_running(loop)
         self._state_lock = threading.RLock()
         self._loop = loop
-
+        if sys.version_info < (3, 8):
+            self._loop_for_kwargs = self._loop
```

Comment:
Random thought - what if you made `partial`s for the 3 different versions of the functions taking `loop=`?

Reply:
Interesting idea.  That would put many more copies of this version check in the code base though?

I think we should also push this version check back to py37 (as that is when https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop came in which is what makes explicitly passing the loop kwarg around unneeded).

Reply:
I meant more along the lines of:

```python
self._asyncio_event = functools.partial(asyncio.Event, loop=(self._loop if sys.version_info < (3, 7) else None))
```
Perhaps it'd be more confusing 🤷
3.7 makes sense, then.

Reply:
Ah, that makes sense.  Seems 6 of one half-dozen of the other.

I don't think we support changing the loop that the RE is attached to?

Reply:
Supporting that seems like a recipe for disaster.